docs/howto/script_testing.md

```diff
@@ -0,0 +1,209 @@
+
+# Bring up a docker container.
+#
+# The service is described in the test-hits/docker-compose.yml below with
+# its logs in test-hits/logs/generated.log.
+docker_up -profile ${CONFIG_PROFILES}/default -network ${NETWORK_NAME} test-hits
+
+# Add the package resources.
+add_package -profile ${CONFIG_PROFILES}/default
+
+# Add the data stream.
+#
+# The configuration for the test is described in test_config.yaml below.
+add_data_stream -profile ${CONFIG_PROFILES}/default test_config.yaml DATA_STREAM_NAME
+
+# Start the service.
+docker_signal test-hits SIGHUP
+
+# Wait for the service to exit.
+docker_wait_exit -timeout 5m test-hits
+
+# Check that we can see our policy.
+get_policy -profile ${CONFIG_PROFILES}/default -timeout 1m ${DATA_STREAM_NAME}
+cp stdout got_policy.json
+exec jq '.name=="'${DATA_STREAM_NAME}'"' got_policy.json
+stdout true
+
+# Take down the service and check logs for our message.
+docker_down test-hits
+! stderr .
+stdout '"total_lines":10'
+
+# Get documents from the data stream.
+get_docs -profile ${CONFIG_PROFILES}/default -want 10 -timeout 5m ${DATA_STREAM_NAME}
+cp stdout got_docs.json
+
+# Remove the data stream.
+remove_data_stream -profile ${CONFIG_PROFILES}/default ${DATA_STREAM_NAME}
+
+# Uninstall the agent.
+uninstall_agent -profile ${CONFIG_PROFILES}/default -timeout 1m 
+
+# Remove the package resources.
+remove_package -profile ${CONFIG_PROFILES}/default
+
+-- test-hits/docker-compose.yml --
+version: '2.3'
+services:
+  test-hits:
+    image: docker.elastic.co/observability/stream:v0.20.0
+    volumes:
+      - ./logs:/logs:ro
+    command: log --start-signal=SIGHUP --delay=5s --addr elastic-agent:9999 -p=tcp /logs/generated.log
+-- test-hits/logs/generated.log --
+ntpd[1001]: kernel time sync enabled utl
+restorecond: : Reset file context quasiarc: liqua
+auditd[5699]: Audit daemon rotating log files
+anacron[5066]: Normal exit ehend
+restorecond: : Reset file context vol: luptat
+heartbeat: : <<eumiu.medium> Processing command: accept
+restorecond: : Reset file context nci: ofdeFin
+auditd[6668]: Audit daemon rotating log files
+anacron[1613]: Normal exit mvolu
+ntpd[2959]: ntpd gelit-r tatno
+-- test_config.yaml --
+input: tcp
+vars: ~
+data_stream:
+  vars:
+    tcp_host: 0.0.0.0
+    tcp_port: 9999
+```
+
+Other complete examples can be found in the [with_script test package](https://github.com/elastic/elastic-package/blob/main/test/packages/other/with_script/data_stream/first/_dev/test/scripts).
+
+
+## Running script tests
+
+The `elastic-package test script` command has the following sub-command-specific
+flags:
+
+- `--continue`: continue running the script if an error occurs
+- `--data-streams`: comma-separated data streams to test
+- `--external-stack`: use external stack for script tests (default true)
+- `--run`: run only tests matching the regular expression
+- `--scripts`: path to directory containing test scripts (advanced use only)
+- `--update`: update archive file if a cmp fails
```

Comment:
Does it mean to update the txtar file itself, for the part with the file?

Reply:
Yes. When there is an e.g. `cmp stdout want.text` and the want.text file in the txtar does not match, it is updated.